app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = [
+        Person(person.get("name"), person.get("age")) for person in people
+    ]
+    i = 0
+    for person in people:
+        if person.get("wife") is not None:
+            result[i].wife = Person.people.get(person["wife"])
+        if person.get("husband") is not None:
+            result[i].husband = Person.people.get(person["husband"])
+        i += 1
```

Comment:
use for i in range(len(people)):
to avoid i = 0, i+= 1

u don't need to use C-like counters in Python